ledger/complete/wal/checkpoint_v6_reader.go

```diff
@@ -520,3 +536,14 @@ func getNodeByIndex(subtrieNodes [][]*node.Node, topLevelNodes []*node.Node, ind
 	// +1 because first item is always nil
 	return topLevelNodes[offset+1], nil
 }
+
+// reachedEOF checks if the reader has reached end of file
+// it returns nil if reached EOF
+// any error returned are exception
+func reachedEOF(reader io.Reader) error {
+	_, err := reader.Read(make([]byte, 1))
+	if errors.Is(err, io.EOF) {
+		return nil
+	}
+	return fmt.Errorf("expect to reach EOF, but didn't: %w", err)
```

Comment:
`err` could be `nil` here.  Maybe just return a sentinel error if EOF isn't reached.